File: container-with-most-water/container-with-most-water.cpp
class Solution {
private:
    int vol{0};
public:
    int maxArea(vector<int>& height) {
        for(vector<int>::iterator left = height.begin(), right = height.end() - 1; left < right; ){
            int min = std::min(*left, *right), distance = std::distance(left, right), area = min * distance;
            vol = (area > vol) ? area : vol ;
            // cout<<vol<<", ";
            if(*left < *right){
                left++;
            } else {
                right--;
            }
        }
        return vol;
    }
};

File: container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right, mx = 0, len(height) - 1, -math.inf
        while left < right:
            mx = max(mx, min(height[left], height[right]) * (right - left))
            if height[left] < height[right]:
                left += 1
            elif height[right] < height[left]:
                right -= 1
            else:
                if height[right - 1] < height[left - 1]:
                    right -= 1
                else:
                    left += 1
            # print(left, right)
        return mx
            
            
        